test/e2e/main_test.go

```diff
@@ -67,53 +62,72 @@ func TestQueryPrometheus(t *testing.T) {
 		{
 			query:   `up{job="node-exporter"} == 1`,
 			expectN: 1,
-		}, {
+		},
+		{
 			query:   `up{job="kubelet"} == 1`,
 			expectN: 1,
-		}, {
-			query:   `up{job="kube-scheduler"} == 1`,
-			expectN: 1,
-		}, {
-			query:   `up{job="kube-controller-manager"} == 1`,
+		},
+		// Not supported in the origin 3.10 topology.
+		// {
+		// 	query:   `up{job="kube-scheduler"} == 1`,
+		// 	expectN: 1,
+		// },
+
+		// Not supported in the origin 3.10 topology.
+		// {
+		// 	query:   `up{job="kube-controller-manager"} == 1`,
+		// 	expectN: 1,
+		// },
+		{
+			query:   `up{job="kube-controllers"} == 1`,
 			expectN: 1,
-		}, {
+		},
+		{
 			query:   `up{job="apiserver"} == 1`,
 			expectN: 1,
-		}, {
+		},
+		{
 			query:   `up{job="kube-state-metrics"} == 1`,
 			expectN: 1,
-		}, {
-			query:   `up{job="prometheus"} == 1`,
+		},
+		{
+			query:   `up{job="prometheus-k8s"} == 1`,
 			expectN: 1,
-		}, {
+		},
+		{
 			query:   `up{job="prometheus-operator"} == 1`,
 			expectN: 1,
-		}, {
+		},
+		{
 			query:   `up{job="alertmanager-main"} == 1`,
 			expectN: 2,
-		}, {
+		},
+		{
 			query:   `namespace:container_memory_usage_bytes:sum`,
 			expectN: 1,
 		},
 	}
 
 	// Wait for pod to respond at queries at all. Then start verifying their results.
-	err := wait.Poll(5*time.Second, 5*time.Minute, func() (bool, error) {
-		_, err := f.QueryPrometheus("prometheus-k8s-0", "up")
+	err := wait.Poll(5*time.Second, 20*time.Second, func() (bool, error) {
+		_, err := client.query("up")
 		return err == nil, nil
 	})
 	if err != nil {
 		t.Fatal(errors.Wrap(err, "wait for prometheus-k8s"))
 	}
 
-	err = wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
+	err = wait.Poll(5*time.Second, 20*time.Second, func() (bool, error) {
 		defer t.Log("---------------------------\n")
 
 		for _, q := range queries {
-			n, err := f.QueryPrometheus("prometheus-k8s-0", q.query)
+			response, err := client.query(q.query)
+			res, err := gabs.ParseJSONBuffer(response.Body)
```

Comment:
We actually do know the format of this response so we should actually parse it correctly and not use this dependency.

Reply:
I wasn't entirely sure why we weren't unmarshaling into a little struct, but I was just doing a quick refactor. Will take care of this too.